delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/jsonld/JsonLdAssembly.scala

```diff
@@ -0,0 +1,78 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.jsonld
+
+import cats.Eq
+import cats.effect.IO
+import cats.syntax.all._
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.RdfError
+import ch.epfl.bluebrain.nexus.delta.sdk.jsonld.JsonLdRejection._
+import ch.epfl.bluebrain.nexus.delta.rdf.graph.Graph
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.api.JsonLdApi
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.{ContextValue, RemoteContext, RemoteContextResolution}
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.{CompactedJsonLd, ExpandedJsonLd}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.jsonld.RemoteContextRef
+import io.circe.Json
+
+/**
+  * Result of the processing of the source from the [[JsonLdSourceProcessor]] which validates that the different
+  * representations and the id is valid or generated
+  * @param id
+  *   the identifier of the resource
+  * @param source
+  *   the original payload
+  * @param compacted
+  *   its compacted json-ld representation
+  * @param expanded
+  *   its expanded json-ld representation
+  * @param graph
+  *   its graph representation
+  * @param remoteContexts
+  *   it
+  */
+final case class JsonLdAssembly(
+    id: Iri,
```

Comment:
I find it a bit strange at first sight to move the id here; i.e. if we have a command, we now get the id from the `JsonLdAssembly`.

Reply:
At creation, we already did that if the id was in the payload or if it is generated.
For updates, there is also a check if the id provided in the segment is equal to the one in the payload.
So the idea was instead of extracting it to copy it aside is to just use it directly to make the signature a bit simpler